### PR TITLE
Autoyast merge

### PR DESCRIPTION
**Pull request by psladek**
action #4602
common part (data, tests)

### CODE REVIEW

File: tests/installation/bootloader.pm [hunk 1]
```diff
@@ -210,7 +210,16 @@ sub run() {
 
     my $args = "";
     if ( get_var("AUTOYAST") ) {
-        $args .= " netconfig=dhcp,all";
+        if (get_var("VERSION")=~/^\D*10\D*/ ) { # sle10 does not have netconfig,ifcfg
+            $args .= " netsetup=dhcp,all";
+        }
```
Comment (coolo) [inline tests/installation/bootloader.pm]: do we really care for sle10?

File: tests/installation/bootloader.pm [hunk 2]
```diff
@@ -210,7 +210,16 @@ sub run() {
...
+        elsif (get_var("VERSION")=~/^\D*11\D*/ ) { # sle11 does not have ifcfg
+            #$args .= " netconfig=dhcp,all";
+            $args .= " netsetup=dhcp,all";  #hotfix, netconfig seems not to work
```
Comment (coolo) [inline tests/installation/bootloader.pm]: what's wrong with just using netconfig=dhcp,all? as it was before? it's supported by all

File: tests/autoyast/useradd.pm [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    $testapi::username = "bernhard";
+    # $testapi::password = "nots3cr3t";
+    $testapi::password = "root";
+
+
+    type_string("useradd -m -c \"$testapi::realname\" $testapi::username\n");
+    sleep 1;
+    type_string("passwd $testapi::username\n");
+    sleep 1;
+    type_password;send_key "ret";
```
Comment (coolo) [inline tests/autoyast/useradd.pm]: these types with sleep look fragile.

I would recommend validate_script_output and using chpasswd within it

**Comment by coolo** [comment]
I merged a rebased version of this now - which also included the simplified netconfig typing we already had in master

File: tests/installation/bootloader.pm (hunk 2 above)
Comment (psladek) [inline tests/installation/bootloader.pm]: Only that netconfig=dhcp,all does not work on SLES11. Which is the reason for the hotfix commit (up until merge, our autoyast tests were all using netsetup), and probably why all the autoyast tests have been failing on openqa.suse.de

Comment (coolo) [inline tests/installation/bootloader.pm]: I see, but please no VERSION checks in tests. set AUTOYAST_USE_NETSETUP in sle11's main.pm

**Pull request closed without merging**